### DISCUSSION TITLE
2.0.0 Beta release

### DISCUSSION THREAD

**Discussion by AvdLee** [AvdLee/appstoreconnect-swift-sdk#170] [Announcements]
Welcome to the beta release of 2.0.0!

**Comment by alexslee** [comment]
@AvdLee since you mentioned to start a discussion here: I just tried using the current tag of `2.0.0-beta1`, and ran into a dependency error inside `APIProvider` (update: looks like a lot of other files, too lol). It tries to `import Get`, but it's not part of the dependencies list in `Package.swift`. Looks like the dependency was removed, but not the import line? **EDIT**: opened a PR to show the delta: #172

**Reply by AvdLee** [reply]
Ah, good point! The fun of build caches that lie! The `APIProvider` should normally be the only one to do manually, the other files are autogenerated and make use of the CreateAPI configuration which still defines the `import Get`. I'll take care of that one!

How's the beta SDK working for you apart from that?

**Reply by alexslee** [reply]
Really well for the most part! Granted, I have only used a few of the endpoints thus far (exploring putting together some internal tooling for work, to keep an eye on a handful of the App Store Connect endpoints that we need on any given day).  The only thing that I have run into issues with so far is the endpoint for Power + Performance Metrics. The one that you hit by calling this:
```swift
let request = APIEndpoint.apps.id(appID).perfPowerMetrics.get(parameters: .init(
      filterDeviceType: ["all_iphones", "all_ipads"],
      filterMetricType: [.battery, .launch, .memory]
    ))
```
Currently, it was auto-generated to just return a non-decoded `Data` type, and the actual format of the response doesn't actually match the only thing I could find that might have been a candidate in the generated set of entities, which is `PerfPowerMetricsResponse` (unless I just blanked and missed the right struct? lol). In any event, I was able to just reference [the Apple docs](https://developer.apple.com/documentation/appstoreconnectapi/get_power_and_performance_metrics_for_an_app), which provided a sample response, to quickly put together a few structs that did the trick.

**Reply by AvdLee** [reply]
Interesting, that would mean the OpenAPI specs are wrong since that's the input for auto-generated code 🤔 

Good to hear most of the SDK works as expected at least, that's a good start!

**Reply by MortenGregersen** [reply]
I have already reported feedback to Apple back in March, about the wrong type in the OpenAPI spec 😣 My feedback has this ID: FB9963088

**Reply by MortenGregersen** [reply]
It looks like Apple has fixed the `xcodeMetrics` issue with version 2.0 of the OpenAPI spec that came out today 🎉

**Reply by alexslee** [reply]
Oh interesting, I just tried grabbing the JSON [from Apple's site](https://developer.apple.com/sample-code/app-store-connect/app-store-connect-openapi-specification.zip), and running `create-api generate app_store_connect_api_2.0_openapi.json --module "AppStoreConnect_Swift_SDK" --output
. --config .create-api.yml -s`, but ran into the following log:
[createapi-output.txt](https://github.com/AvdLee/appstoreconnect-swift-sdk/files/9088781/createapi-output.txt)

with no actual changes to commit after having run the command. Not sure if I just did something wrong though, first time using CreateAPI 😅 (pulled the latest source from the repo + set it up directly via `make install`).

**Reply by AvdLee** [reply]
@alexslee I've already got that fixed locally. I'm pushing out a release tonight after work!
